src/main/java/org/folio/innreach/domain/service/impl/RecordContributionServiceImpl.java

```diff
@@ -0,0 +1,122 @@
+package org.folio.innreach.domain.service.impl;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.stereotype.Service;
+
+import org.folio.innreach.domain.service.ContributionValidationService;
+import org.folio.innreach.domain.service.RecordContributionService;
+import org.folio.innreach.dto.Holding;
+import org.folio.innreach.dto.Instance;
+import org.folio.innreach.dto.Item;
+import org.folio.innreach.external.service.InnReachExternalService;
+
+@Log4j2
+@AllArgsConstructor
+@Service
+public class RecordContributionServiceImpl implements RecordContributionService {
+  private final ContributionValidationService validationService;
+  private final InnReachExternalService innReachExternalService;
+
+  @Override
+  public Set<UUID> evaluateInventoryItemForContribution(Item item, Set<UUID> centralServerIds) {
+    log.info("Evaluating inventory item.");
+    Set<UUID> notValidCodes = new HashSet<>();
+    var statisticalCodeIds = item.getStatisticalCodeIds();
+    var holdingStatisticalCodes = item.getHoldingStatisticalCodeIds();
+
+    if (statisticalCodeIds.size() != 1 || holdingStatisticalCodes.size() != 1) {
+      log.warn("Multiple statistical codes defined, item with id = {} is to be decontributed.", item.getId());
+      notValidCodes = centralServerIds;
+    } else {
+      for (UUID code : centralServerIds) {
+        if (!validStatisticalCode(code, statisticalCodeIds)) {
+          log.warn("Item with id = {}  has \"Do not contribute\" statistical code, it is to be decontributed" +
+            "from central server with code = {}.", item.getId(), code);
+          notValidCodes.add(code);
+        } else if (!validStatisticalCode(code, item.getHoldingStatisticalCodeIds())) {
+          log.warn("Parent holding of item with id = {} has \"Do not contribute\" statistical code, item is to be decontributed" +
+            "from central server with code = {}.", item.getId(), code);
+          notValidCodes.add(code);
+        }
+      }
+    }
+    log.info("Inventory item evaluation complete.");
+    return notValidCodes;
+  }
+
+  private boolean validStatisticalCode(UUID centralServerCode, List<UUID> statisticalCodes) {
+    var suppressionCode = validationService.getSuppressionStatus(centralServerCode, statisticalCodes);
+    return suppressionCode == null || !suppressionCode.equals('n');
+  }
+
+  @Override
+  public void decontributeInventoryItemEvents(Item item, Set<UUID> centralServerCodes) {
+    log.info("Decontributing item with id = " + item.getId());
+
+    centralServerCodes.forEach(code ->
+      innReachExternalService.deleteInnReachApi(code.toString(), "/contribution/item/" + item.getId()));
```

Comment:
item is linked by hrid, not by id. 
Check the following doc, it might be outdated a bit but still should be useful to understand  FOLIO to D2IR records mapping https://wiki.folio.org/display/FOLIJET/SPIKE%3A+%5BMODINREACH-79%5D+Record+Contribution%3A+Analyze+data+required+for+D2IR+contribution